aspnetcore/fundamentals/middleware/index.md

```diff
@@ -99,8 +99,8 @@ In the preceding code:
 * Not every middleware appears in this exact order, but many do. For example:
   * `UseCors`, `UseAuthentication`, and `UseAuthorization` must appear in the order shown.
   * `UseCors` currently must appear before `UseResponseCaching`. This requirement is explained in [GitHub issue dotnet/aspnetcore #23218](https://github.com/dotnet/aspnetcore/issues/23218).
-  * `UseRequestLocalization` must appear before any middleware that might check the request culture, for example, `app.UseMvcWithDefaultRoute`.
-  * <xref:Microsoft.AspNetCore.Builder.RateLimiterApplicationBuilderExtensions.UseRateLimiter%2A> must be called after `UseRouting` when rate limiting endpoint specific APIs are used. For example, if the [`[EnableRateLimiting]`](xref:Microsoft.AspNetCore.RateLimiting.EnableRateLimitingAttribute) Attribute is used, `UseRateLimiter` must be called after `UseRouting`.
+  * `UseRequestLocalization` must appear before any middleware that might check the request culture, for example, `app.MapControllerRoute`.
```

Comment:
MapControllerRoute isn't middleware. It configures routing.

UseRouting could certainly execute endpoints that use localization. It could be used as an example, but minimal host usually automatically adds it for you.

Reply:
> MapControllerRoute isn't middleware. It configures routing.

Arg, either is `UseMvcWithDefaultRoute`, and it's not an old API either.

Reply:
`UseMvcWithDefaultRoute` is registering middleware.

It hasn't been obsoleted, but `MapDefaultControllerRoute` is the replacement that people should be using.